src/components/clusterDetail/ClusterDetailStatusVarieties.tsx

```diff
@@ -0,0 +1,122 @@
+import React from 'react';
+import { GridItem } from '@patternfly/react-core';
+import {
+  Cluster,
+  Credentials,
+  getClusterCredentials,
+  MonitoredOperator,
+  MonitoredOperatorsList,
+} from '../../api';
+import { getOlmOperators } from '../clusters/utils';
+import ClusterCredentials from './ClusterCredentials';
+import ClusterInstallationError from './ClusterInstallationError';
+import FailedHostsWarning from './FailedHostsWarning';
+import FailedOperatorsWarning from './FailedOperatorsWarning';
+import { getClusterDetailId } from './utils';
+
+type ClusterStatusVarieties = {
+  credentials?: Credentials;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  credentialsError?: any;
+  olmOperators: MonitoredOperatorsList;
+  failedOlmOperators: MonitoredOperatorsList;
+  consoleOperator?: MonitoredOperator;
+  fetchCredentials: () => void;
+};
+
+export const useClusterStatusVarieties = (cluster: Cluster): ClusterStatusVarieties => {
+  const [credentials, setCredentials] = React.useState<Credentials>();
+  const [credentialsError, setCredentialsError] = React.useState();
+
+  const olmOperators = getOlmOperators(cluster.monitoredOperators);
+  const failedOlmOperators = olmOperators.filter((o) => o.status === 'failed');
+  const consoleOperator = React.useMemo(
+    () => cluster.monitoredOperators?.find((o) => o.name === 'console'),
+    [cluster.monitoredOperators],
+  );
+
+  const fetchCredentials = React.useCallback(() => {
+    const fetch = async () => {
+      setCredentialsError(undefined);
+      try {
+        const response = await getClusterCredentials(cluster.id);
+        setCredentials(response.data);
+      } catch (err) {
+        setCredentialsError(err);
+      }
+    };
+    fetch();
+  }, [cluster.id]);
+
+  React.useEffect(() => {
+    if (
+      (!consoleOperator && cluster.status === 'installed') || // Retain backwards compatibility with clusters which don't have monitored clusters
+      consoleOperator?.status === 'available'
+    ) {
+      fetchCredentials();
+    }
+  }, [cluster.status, consoleOperator, fetchCredentials]);
+
+  return {
+    credentials,
+    credentialsError,
+    olmOperators,
+    failedOlmOperators,
+    consoleOperator,
+    fetchCredentials,
+  };
+};
+
+const ClusterDetailStatusVarieties: React.FC<{
+  cluster: Cluster;
+  setResetClusterModalOpen: (isOpen: boolean) => void;
+  clusterVarieties: ClusterStatusVarieties;
```

Comment:
How about calling the hook directly in this component rather than doing it in parent and passing it in as prop? It is not used in the parent in any way.

Reply:
Actually it is - in the ClusterDetail flow, the credentials are passed to ClusterCredentials. Querying it in the parent component is one of the ways to ensure just a single query per component hierarchy.
We can wrap iit n a context or do another refactoring of the hierarchy, but the `ClusterDetailStatusVarieties` is not expected to be reused any farther in the future so this is the most straightforward way, imo.

Reply:
So you're expecting the `useClusterStatusVarieties` hook to be reused somewhere else other than to populate data for `ClusterDetailStatusVarieties` component?

The hook is preparing data specific for the `ClusterDetailsStatusVarieties` component. I think trying to make it reusable is wrong.